tests/test_extensions/test_ext_apidoc.py

```diff
@@ -53,6 +55,28 @@ def test_simple(make_app, apidoc):
     print(app._warning.getvalue())
 
 
+def test_custom_templates(rootdir):
```

Comment:
Given the new simplified test this could go into `test_util/test_util_template.py` instead...

Reply:
I think it's useful to have coverage here as confirmation that the reported regression has been solved; perhaps some unit testing on the template code does make sense too though.

And yep, almost every time I go to modify/add a testroot I question whether it is a good idea or not.  In a way I think it's nice to have a compendium of examples of "this is how we think a given project should be displayed" - engineering-wise though, that seems unwieldly (currently?) and so I'll try to use unit testing more often.

Reply:
To provide more-actionable feedback, though: yep, if we keep this test as-is, I do think it should probably be relocated to `test_util_template.py` (or even a `test_jinja2glue.py`?) since it's exercising code that is only coincidentally `apidoc`-related.

Reply:
Oh, missed this sorry. Feel free to update as you see fit & ping me when done!

A

Reply:
Ok, I've started looking at restructuring the test.

  * :white_check_mark: I like that the templates are simplified - no need to keep the irrelevant template content.
  * :white_check_mark: Adding the assertion on the directory listing is sensible; it confirms that the inputs are as expected, and that a successful result isn't due to some other unrelated content.
  * :x: I'm not so keen on the updated `testroot` - it's not really a complete Sphinx project root, and I don't think that's in the spirit of the testroots.
  * After more consideration I do think that the test -- because it is a regression fix -- should stay in the `apidoc` part of the test suite.

I'll try to find a middle-ground that accommodates these.

Reply:
> I'll try to find a middle-ground that accommodates these.

I'm not sure the changes I've pushed are ideal; some of the test framework is still a bit quirky.  But I think it is cleaner than it was originally.